### PR TITLE
838 fix displayed id

### LINKED ISSUE #838: ID displayed isn't good when document include a field "id"
## Context

Hi,
I don't know if it's a feature or a bug but I spent a lot of time to understand what happens.

It's simple, when documents include a field "id", the console use this field as identifier in view line and not real document id.
In my usage I generate a specific ID for documents to get them quickly and "id" field in my document store other information.

So, as I said, I don't know if it's a bug or a feature, but I lost time to find solution and it's when I used directly API that I saw that ID was ok after my imports. 

## Expected Behavior
See view line of documents with good document _id

## Current Behavior

We saw field "id" in document and not reald document _id

## Possible Solution

To "fix" this I rename field id and reimport documents

## Steps to Reproduce

1. Import document this a field "id"
2. Show them in Console next generation

## Context (Environment)

Kuzzle version: 2.6.0
Node.js version: v12.13.1
Admin Console Version: 4-dev (http://next-console.kuzzle.io/)

### PR DESCRIPTION
**Pull request by berthieresteban**
## What does this PR do ?
fix #838 

### How should this be manually tested?
  - Step 1 : create a collection with field `id` in mapping
  - Step 2 : create a document with a value in `id` field
  - Step 3 : check on document views (list/column/map...), the _id is now  displayed
  ...

### Other changes
disable map view when mapping have no `geo_point` field
disable chart view when mapping have no `integer` field

### Boyscout
catch json editor error (browser console was spammed when creating/editing documents)

### CODE REVIEW

**Comment by berthieresteban** [comment]
Draft -> I forgot to write some tests

File: src/components/Data/Documents/Common/CreateOrUpdate.vue [hunk 1]
```diff
@@ -196,7 +197,13 @@ export default {
   methods: {
     onJsonChange(val) {
       this.rawDocument = val
-      this.$emit('document-change', JSON.parse(val))
+      let parsed = {}
+      try {
+        parsed = JSON.parse(val)
+        this.$emit('document-change', parsed)
+      } catch (error) {
+        // invalid json
```
Comment (alexandrebouthinon) [inline src/components/Data/Documents/Common/CreateOrUpdate.vue:205]: Maybe we should do something in case of error. At least log it into the JS console

**Review by alexandrebouthinon — APPROVED** [review]

File: src/components/Data/Documents/Common/CreateOrUpdate.vue (hunk 1 above)
Comment (berthieresteban) [reply src/components/Data/Documents/Common/CreateOrUpdate.vue:205]: That method is called every time the text in json editor changed, I'm not sure we want to print an error in console when user is writing something and haven't finished so the json can't be parsed :/
For example, just write `{"test":"test"}` will print at least 7 error in console

Comment (berthieresteban) [reply src/components/Data/Documents/Common/CreateOrUpdate.vue:205]: That method is called every time the text in json editor changed, I'm not sure we want to print an error in console when user is writing something and haven't finished so the json can't be parsed :/
For example, just write `{"test":"test"}` will print at least 7 error in console

Comment (Leodau) [inline src/components/Data/Documents/Common/CreateOrUpdate.vue:205]: If we do not want to log or do anything with this error, then add an explicit comment.
```suggestion
        // Fail silently
```

**Review by Leodau — APPROVED** [review]
This bug was killing me :muscle:, clean fix.
I will still wait for someone from the AC team to merge, tho.

**Review by xbill82 — APPROVED** [review]
🎉

**Pull request merged**